id.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2014 - Max Ekman <max@looplab.se>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eventhorizon
+
+import (
+	"fmt"
+
+	"gopkg.in/mgo.v2/bson"
+)
+
+// ID is an unique identifier type
+type ID struct {
```

Comment:
I do have some plans on providing a default implementation of Event to use in most cases instead of the domain implementations, those structs would instead be data for the event. See https://github.com/looplab/eventhorizon/issues/39. That change would at least make the pluggable ID types easier to contain as domain types would not have any event/command ids, instead they are contained internally in the default Event implementation. Do you think those changes first would make implementing the pluggable ID types easier?